spec/abstractops.html

```diff
@@ -1213,10 +1213,10 @@ <h1>ISO 8601 grammar</h1>
       TemporalZonedDateTimeString :
           Date TimeSpecSeparator? TimeZoneNameRequired Calendar?
 
-      TemporalCalendarString :
-          CalendarName
+      ISODateTimeString :
           TemporalInstantString
-          CalendarDateTime
+          TemporalDateTimeString
+          TemporalZonedDateTimeString
```

Comment:
I'm not sure whether there's still ambiguity in ISODateTimeString at this point. But if there is, that's probably something we can solve without a normative change, since unlike the rest of this change it won't affect whether a string is parsed as an ISO date or calendar name.

Reply:
I changed my mind about this because of #2379 — I figured that when we allow calendar annotations after DateSpecYearMonth and DateSpecMonthDay, we'll probably have to treat them differently by throwing on non-ISO calendar annotations. (that is, `2022-08-31[u-ca=chinese]` is allowed, in the CalendarDateTime branch of ISODateTimeString, but `08-31[u-ca=chinese]`, in the DateSpecMonthDay branch, is not). This would be difficult to do, if we collapsed this loop into the ISODateTimeString production. So I think it's probably better not to introduce ISODateTimeString in this PR.

In that case, we could just keep the for-each step, and delete TemporalCalendarString altogether.

We could investigate adding ISODateTimeString later (possibly excluding YearMonth and MonthDay) in an editorial PR.